wire-ios-data-model/Source/MLS/MLSService.swift

```diff
@@ -1386,6 +1382,8 @@ public final class MLSService: MLSServiceInterface {
                         logger.error("failed to commit pending proposals: \(String(describing: error))")
                     }
                 }
+
+                await taskGroup.waitForAll()
```

Comment:
Hmm I am not really sure how to understand this. In the documentation it says for `withTaskGroup(of:returning:body:)`:
> A group waits for all of its child tasks to complete or be canceled before it returns. After this function returns, the task group is always empty.

So why is this `await taskGroup.waitForAll()` needed?

Reply:
1. I agree with Thomas, the `waitForAll()` is not required, if there are no statements in the block afterwards
2. More importantly, the `waitForAll()` is execute on each iteration of the loop, right after the `.addTask {}` call. So either it is placed in the wrong line here or there is no point in having a task group at all, since there will not be any parallel execution (`waitForAll` right after `add`)

Reply:
1. In theory, I agree with what the documentation says. But in practice it doesn't seem to work like that: without waiting for all tasks, the tasks simply do not complete before `await commitPendingProposals()` finishes executing. [This example](https://www.avanderlee.com/concurrency/task-groups-in-swift/) also suggests that to collect all values returned by a task group, one musts `await` for them all to return.

2. Good catch. I meant to do that outside of the loop

Reply:
@David-Henner do you have a way to reproduce the flakiness to validate?
When I run the test 200 times it always passes, if the line `await taskGroup.waitForAll()` exists or not. So maybe there are other conditions that make it flake, but I would question this solution.

Reply:
If I run the test in isolation it systematically fails

Reply:
For me it doesn't 🤯
```
Test Case '-[WireDataModelTests.MLSServiceTests test_CommitPendingProposals_ForSubconversation]' passed (0.060 seconds).
Test Suite 'MLSServiceTests' passed at 2024-03-21 10:51:29.574.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.060 (0.060) seconds
Test Suite 'WireDataModelTests.xctest' passed at 2024-03-21 10:51:29.574.
	 Executed 1 test, with 0 failures (0 unexpected) in 0.060 (0.060) seconds
```

Did you try to reset your simulator and clean derived data and so on? Do you have any other setup than me, like an Intel processor instead of ARM based chip?

Reply:
discussed with @markusfassbender and updated https://github.com/wireapp/wire-ios/pull/1153/commits/40b46ed2b876d8adcba28f8d9a5ca591d6a5d88e